src/esi/ExpatParser.cc

```diff
@@ -10,7 +10,7 @@
 
 #include "squid.h"
 
-#if USE_SQUID_ESI && HAVE_LIBEXPAT
+#if HAVE_LIBEXPAT
```

Comment:
Here and elsewhere in this PR, I am against removing USE_SQUID_ESI precondition (even if HAVE_LIBEXPAT implies USE_SQUID_ESI today). If some ESI code is using USE_SQUID_ESI, then all ESI code should be using USE_SQUID_ESI (to the extent possible).

Reply:
Since the removal of the custom ESI parser `USE_SQUID_ESI` has become redundant with the two library `HAVE_FOO` macros. I have now gone the whole way to full removal of the `--disable-esi` build option.